oximeter/collector/src/lib.rs

```diff
@@ -157,6 +159,49 @@ pub struct OximeterArguments {
     pub address: SocketAddrV6,
 }
 
+// Provides an alternative to the DNS resolver for cases where we want to
+// contact a backend without performing resolution.
+struct SingleHostResolver {
+    tx: watch::Sender<AllBackends>,
+}
+
+impl SingleHostResolver {
+    fn new(address: SocketAddr) -> Self {
+        let backends = Arc::new(BTreeMap::from([(
+            backend::Name::new("singleton"),
+            backend::Backend { address },
+        )]));
+        let (tx, _rx) = watch::channel(backends.clone());
+        Self { tx }
+    }
+}
+
+impl Resolver for SingleHostResolver {
+    fn monitor(&mut self) -> watch::Receiver<AllBackends> {
+        self.tx.subscribe()
+    }
+}
+
+// A "qorb connector" which converts a SocketAddr into a nexus_client::Client.
+struct NexusConnector {
+    log: Logger,
+}
+
+#[async_trait::async_trait]
+impl backend::Connector for NexusConnector {
```

Comment:
The full `Connector` interface is defined here: https://docs.rs/qorb/0.0.1/qorb/backend/trait.Connector.html

I would actually like to make an `is_valid` implementation that's smarter here, but I'm not sure how to do that generically on top of a [reqwest::Client](https://docs.rs/reqwest/0.12.8/reqwest/struct.Client.html)... I suppose I could make a `GET` method to Nexus, but it would be nice to have a way to check the TCP connection without actually sending a query.

Reply:
Yeah, I don't think that is really feasible using `reqwest`, since you don't have access to the actual TCP socket it's using. I'd not worry about it too much here, it only needs to look up Nexus very rarely.